languages/Language.php

```diff
@@ -137,18 +137,30 @@ class Language {
 
 	/**
 	 * Get a cached language object for a given language code
+	 * !!!Wikia changed - entire method!!!
+	 *
 	 * @param $code String
+	 * @param $cityId int - used only in global title to force rebuilding cache for other wiki
 	 * @return Language
 	 */
-	static function factory( $code ) {
-		if ( !isset( self::$mLangObjCache[$code] ) ) {
+	static function factory( $code, $cityId = null ) {
+		global $wgCityId;
+		if ( is_null( $cityId )) {
+			$cityId = $wgCityId;
+		}
+
+		if ( !isset( self::$mLangObjCache[$code][$cityId] ) ) {
 			if ( count( self::$mLangObjCache ) > 10 ) {
 				// Don't keep a billion objects around, that's stupid.
 				self::$mLangObjCache = array();
```

Comment:
Do we want to keep it or believe it's stupid to clean the cache like that? If we want to keep then let's make it `= []`